lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -2399,10 +2408,28 @@ LogicalResult StmtEmitter::visitStmt(InstanceOp op) {
 
     // Emit the value as an expression.
     ops.clear();
-    // output ports were lowered to wire
-    if (elt.isOutput())
+    // output ports that are not connected to single use output ports were lowered to wire
+    OutputOp output = dyn_cast_or_null<OutputOp>(portVal.getUses().begin()->getOwner());
+    if (elt.isOutput() && portVal.hasOneUse() && output) {
+      auto module = output->getParentOfType<HWModuleOp>();
+      SmallVector<ModulePortInfo> portInfo = getModulePortInfo(module);
+
+      for (size_t portNum = 0, operandNum = 0; portNum < portInfo.size(); ++portNum) {
+        if (!portInfo[portNum].isOutput())
+          continue;
+
+        if (output->getOperand(operandNum) == portVal) {
+          os << portInfo[portNum].getName();
+          break;
+        }
+
+        ++operandNum;
+      }
+
+    } else if (elt.isOutput()) {
```

Comment:
I'd recommend pulling the "isOutput" higher since it encompases both checks.  You could even invert this logic a bit to be more readable, something like:

```
if (!elt.isOutput()) {
  emitExpression(portVal, ops);
} else if (<not the inline case>) {
  emitExpression(getWireForValue(portVal), ops);
} else {
  ...
  os << the name
}
```